### ISSUE TITLE
Program without relocations leads to crash

### ISSUE THREAD

**Issue by neuschaefer** [fox32-arch/fox32os#4]
When I build a FXF binary and load it using `fox32 --disk fox32is.img --disk foo.fxf`, some binaries lead to a crash:

```
; works
a: jmp    a
```

This works.

```
; breaks
a: rjmp    a
```

This throws me into the fox32rom monitor, with the messages:

```
Page fault at virtual address r1
Breakpoint reached!
```

**Comment by ry755** [comment]
Oh I think I know what's going on here. In kernel/fxf/reloc.asm, it doesn't do a zero check before looping on the count of relocations. This has the effect of looping the 32 bit max integer number of times because the loop counter underflows from 0. oopsies

**Comment by ry755** [comment]
Fixed in e577446e555036f955130df22ab0f91463f44463.
